### PR TITLE
Publish ECDH certificate from TPM mgr to Zedagent

### PR DESCRIPTION
**Pull request by zed-rishabh**
Added support of publishing ECDH certificate from the TPM manager to the Zedagent which will be used for the encryption by the controller. The missing part in this PR is publishing the certificate to the controller from the zedagent after receiving them. I had a discussion with @srinibas-zededa , he will take care of it in a separate PR.

### CODE REVIEW

**Comment by zed-rishabh** [comment]
@cshari-zededa Please take a look.

File: pkg/pillar/cmd/zedagent/handlecertconfig.go [hunk 1]
```diff
@@ -193,3 +193,19 @@ func unpublishControllerCert(ctx *getconfigContext, key string) {
...
+	return
+}
+
+func handleAttestCertDelete(ctxArg interface{}, key string,
+	configArg interface{}) {
+
+	status := configArg.(types.AttestCert)
+	log.Infof("handleAttestCertDelete for %s\n", status.Key())
+	return
+}
```
Comment (eriknordmark) [inline pkg/pillar/cmd/zedagent/handlecertconfig.go:211]: Are you/Srinibas planning to hook in code here later? If so might make sense to add some XXX TBD comment about it.

File: pkg/pillar/cmd/tpmmgr/tpmmgr.go [hunk 2]
```diff
@@ -999,6 +1000,63 @@ func createEcdhCert() error {
...
+	return certBytes, nil
+}
+
+func getCertHash(cert []byte, hashAlgo types.CertHashType) ([]byte, error) {
+	certHash := sha256.Sum256(cert)
+	switch hashAlgo {
+	case types.CertHashTypeSha256First16:
+		return certHash[:16], nil
+	default:
+		return []byte{}, fmt.Errorf("Unsupported cert hash type")
```
Comment (eriknordmark) [inline pkg/pillar/cmd/tpmmgr/tpmmgr.go:1028]: How about adding a %d to include the hashAlgo in the error message?

File: pkg/pillar/cmd/zedagent/handlecertconfig.go (hunk 1 above)
Comment (zed-rishabh) [reply pkg/pillar/cmd/zedagent/handlecertconfig.go:211]: @srinibas-zededa is planning to hook in code here. I will add the necessary comments.

File: pkg/pillar/cmd/tpmmgr/tpmmgr.go (hunk 2 above)
Comment (zed-rishabh) [reply pkg/pillar/cmd/tpmmgr/tpmmgr.go:1028]: Yeah, it makes sense. I will add the hashAlgo in the error message.

**Comment by zed-rishabh** [comment]
> LGTM. Two nits.

I updated the PR. Thanks!

**Pull request merged**